tests/Teensy_ECDSA_Sign/Teensy_ECDSA_Sign.ino

```diff
@@ -0,0 +1,148 @@
+/*
+  Using the SparkFun Cryptographic Co-processor Breakout ATECC508a (Qwiic)
+  By: Pete Lewis
+  SparkFun Electronics
+  Date: August 5th, 2019
+  License: This code is public domain but you can buy me a beer if you use this and we meet someday (Beerware license).
+
+  Feel like supporting our work? Please buy a board from SparkFun!
+  https://www.sparkfun.com/products/15573
+
+  This example shows how to create a digital signature (ECC type) on 32 bytes of data.
+  Note, this requires that your device be configured with SparkFun Standard Configuration settings.
+  By default, this example uses the private key securely stored and locked in slot 0.
+
+  Hardware Connections and initial setup:
+  Install artemis in boards manager: http://boardsmanager/All#Sparkfun_artemis
+  Plug in your controller board (e.g. Artemis Redboard, Nano, ATP) into your computer with USB cable.
+  Connect your Cryptographic Co-processor to your controller board via a qwiic cable.
+  Select TOOLS>>BOARD>>"SparkFun Redboard Artemis"
+  Select TOOLS>>PORT>> "COM 3" (note, yours may be different)
+  Click upload, and follow configuration prompt on serial monitor at 115200.
+
+*/
+
+#include <SparkFun_ATECCX08a_Arduino_Library.h> //Click here to get the library: http://librarymanager/All#SparkFun_ATECCX08a
+#include <i2c_t3.h>
+
+ATECCX08A atecc;
+
+// array to hold our 32 bytes of message. Note, it must be 32 bytes, no more or less.
+uint8_t message[32] = {
+  0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,
+  0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F
+};
+
+void setup() {
+  while(!Serial);
+  Wire.begin(I2C_MASTER, 0x00, I2C_PINS_18_19, I2C_PULLUP_EXT, 100000);
+  Serial.begin(115200);
+  if (atecc.begin() == true)
+  {
+    Serial.println("Successful wakeUp(). I2C connections are good.");
+  }
+  else
+  {
+    Serial.println("Device not found. Check wiring.");
+    while (1); // stall out forever
+  }
+
+  printInfo(); // see function below for library calls and data handling
+
+  // check for configuration
+  if (!(atecc.configLockStatus && atecc.dataOTPLockStatus && atecc.slot0LockStatus))
+  {
+    Serial.print("Device not configured. Please use the configuration sketch.");
+    while (1); // stall out forever.
+  }
+
+  printMessage(); // nice debug to see what you're signing. see function below
+
+  //Let's create a digital signature!
+  atecc.createSignature(message); // by default, this uses the private key securely stored and locked in slot 0.
+
+  printSignature();
+}
+
+void loop()
+{
+  // do nothing.
+}
+
+void printMessage()
+{
+  Serial.println("uint8_t message[32] = {");
+  for (int i = 0; i < sizeof(message) ; i++)
+  {
+    Serial.print("0x");
+    if ((message[i] >> 4) == 0) Serial.print("0"); // print preceeding high nibble if it's zero
+    Serial.print(message[i], HEX);
+    if (i != 31) Serial.print(", ");
```

Comment:
I think this can be rearranged to: 
if ((31 - i) % 16 != 0) {
Serial.print(", ");
} else {
Serial.println();
}
but it's only getting rid of one "if" check so not much would be saved